shared/reducers/route-tree.js

```diff
@@ -15,6 +15,50 @@ import {
 
 const initialState = Constants.State()
 
+function routeChangedReducer(routeChanged, action) {
+  const newRouteChanged = (function() {
```

Comment:
Note to self: clean up this function.

Reply:
So i'm wondering, is this all needed? Is the logic like we have a couple of phases that define what should happen, like
1. Loading (don't store any route changes)
2. Setting up (don't story any route changes)
3. Loaded (start saving route changes)

then then we just watch switchTo   / navigateTo and when that action runs we basically fire off an action to store our state?

Reply:
That's it, except we don't have to explicitly fire off an action to store our state, since the prop change triggered by the switchTo / navigateTo that flips the switch to start saving route changes will already trigger _persistRoute in Main.